### DISCUSSION TITLE
`404 - Could not resolve webview id` Returned on indirect file request

### DISCUSSION THREAD

**Discussion by stneveadomi** [microsoft/vscode-discussions#1136] [Extension Development QnA] [answered]
Hello all! 

I am having an issue loading a .ttf file from a webview in an extension I am developing. 

![image](https://github.com/microsoft/vscode-discussions/assets/28165093/62a42db9-7ef0-4647-9298-63e09a751b8e)

Has anyone seen this issue before?

The TTF fails to load when it is indirectly requested.
For example,
This initiator chain succeeds in fetching the asset:
![image](https://github.com/microsoft/vscode-discussions/assets/28165093/8619ec88-d2ce-40fc-b777-7f71755de918)

But when I load the TTF file via a stylesheet, I receive the 404 error as well as the missing webview client id message in the console:
![image](https://github.com/microsoft/vscode-discussions/assets/28165093/49e63185-5638-46f5-a4e6-de4af7514cf4)

I can guarantee that the file exists. It seems from some preliminary debugging that for some reason it is having trouble finding the **webview client id** when requesting the asset?

## Further debugging
It seems that this url is missing the query param for ID?
![image](https://github.com/microsoft/vscode-discussions/assets/28165093/04de0608-785d-4b2e-bf4d-3d10a3196a53)

![image](https://github.com/microsoft/vscode-discussions/assets/28165093/fa0c8bae-878d-4750-9360-ebdd0d64fd86)

Meanwhile a successful file access has the ID query parameter.
![image](https://github.com/microsoft/vscode-discussions/assets/28165093/245811a2-78e9-4d76-9b4a-0955dee46031)

Seems weird.

**Comment by tamuratak** [comment]
The following example of loading [codicons](https://github.com/microsoft/vscode-codicons) in a [webview](https://code.visualstudio.com/api/extension-guides/webview) works for me. If the example doesn't work on your machine, you had better report to  the [official repository](https://github.com/Microsoft/vscode).

- https://github.com/microsoft/vscode-extension-samples/tree/main/webview-codicons-sample

**Reply by stneveadomi** [reply]
Thank you for the follow up. I have tested and compared this repository to my own, and sadly, I am able to run the example repository fine, but not my own.

I'll look into reporting this in the official repository.

I think the only notable difference between my code and the example code is I am using a WebViewProvider instead of a WebViewPanel which eventually loads an Angular SPA.

**Comment by stneveadomi** [comment, marked as answer]
Update:
Solved the issue.

I can't quite explain why, but the 404s disappeared when I added this `<base>` tag to my `<head>` block.

Hope this helps others
```
//where extensionUri is the extension uri provided to the webview.
<base href="${this._extensionUri}">
```

**Reply by stneveadomi** [reply]
Assumably, the webview somehow was making requests under the wrong base href, hence the resolve webview id error. Adding this default forces all requests to go under that same href.
